src/main/java/com/incentives/piggyback/offers/serviceImpl/OfferServiceImpl.java

```diff
@@ -38,45 +42,35 @@
 	@Autowired
 	private Environment env;
 
+	@Autowired
+	private OfferRepository offerRepository;
+
+	Gson gson = new Gson();
+
 	@Override
-	public ResponseEntity<OfferDTO> updateOfferStatus(Long id, OfferDTO offers)  {
-		updateOffer(offers);
-		//PUSHING MESSAGES TO GCP
-		messagingGateway.sendToPubsub(
-				CommonUtility.stringifyEventForPublish(
-						offers.toString(),
-						Constant.OFFER_CREATED_EVENT,
-						Calendar.getInstance().getTime().toString(),
-						"",
-						Constant.OFFER_SOURCE_ID
-						));
-		return ResponseEntity.ok(offers);
+	public void offerForPartnerOrder(PartnerOrderDTO partnerOrderDTO) {
+		OfferEntity offerEntity = offerRepository.save(ObjectAdapter.generateOfferEntity(partnerOrderDTO));
+		publishOffer(offerEntity, Constant.OFFER_CREATED_EVENT);
+		List<Long> usersList = getNearbyUsers(offerEntity.getInitiatorUserId(), offerEntity.getOrderLocation().getLatitude(),
```

Comment:
Can rename usersList to userIdsList for clarity

Reply:
done!